libs/server/repository/core/application-services/src/lib/read-models/factories/single-repository-data-read-model.factory.ts

```diff
@@ -0,0 +1,23 @@
+import { RepositoryEntity } from '@pimp-my-pr/server/repository/core/domain';
+import { ListRepositoriesReadModel } from '@pimp-my-pr/server/repository/core/application-services';
+import { SingleRepositoryDataReadModel } from '../../queries/get-single-repository-data/single-repository-data.read-model';
+
+export const singleRepositoryDataReadModelFactory = (
+  repository: RepositoryEntity
+): ListRepositoriesReadModel => {
+  if (!repository) {
```

Comment:
I think that this check is redundant until the function arguments don't allow nullable value.